core/input/input.cpp

```diff
@@ -270,6 +271,19 @@ bool Input::is_mouse_button_pressed(MouseButton p_button) const {
 	return mouse_button_mask.has_flag(mouse_button_to_mask(p_button));
 }
 
+void Input::_project_settings_changed() {
+	ignore_joypad_on_unfocused = bool(GLOBAL_GET("input_devices/config/ignore_joypad_on_unfocused_window"));
+}
+
+bool Input::_config_ignore_joypad_on_unfocused_window() {
+	if (ignore_joypad_on_unfocused) {
+		if (!DisplayServer::get_singleton()->window_is_focused(DisplayServer::MAIN_WINDOW_ID)) {
```

Comment:
This approach doesn't cover the case, that an application can have multiple OS-Windows. Perhaps a better approach would be to take application focus instead of window focus into consideration.